test/datasets_utils.py

```diff
@@ -166,9 +166,11 @@ class DatasetTestCase(unittest.TestCase):
 
     Optionally, you can overwrite the following class attributes:
 
-        - CONFIGS (Sequence[Dict[str, Any]]): Additional configs that should be tested. Each dictonary can contain an
+        - CONFIGS (Sequence[Dict[str, Any]]): Additional configs that should be tested. Each dictionary can contain an
             arbitrary combination of dataset parameters that are **not** ``transform``, ``target_transform``,
-            ``transforms``, or ``download``. The first element will be used as default configuration.
+            ``transforms``, or ``download``. Each config will be updated by potentially missing default values.
+        - DEFAULT_CONFIG (Dict[str, Any]): Config that will be used by default. This defaults to all keyword arguments
```

Comment:
"Config that will be used by default": I might be missing something but is this really the case? As mentionned above it doesn't seem that the `DEFAULT_CONFIG` actually gets tested.

If I'm correct and if we keep the current logic, it might be worth renaming `DEFAULT_CONFIG` to "DEFAULT_PARAMETER_VALUES" or something like that

Reply:
The `DEFAULT_CONFIG` is used every time we call `create_dataset` without the `config` parameter:

https://github.com/pytorch/vision/blob/c00c4d36d07552dbe0cc4fd399a9bf911bf99fcb/test/datasets_utils.py#L311-L314

If a config is specified, we overwrite the default config. This is what happens in every built-in test that uses the `@test_all_configs` decorator:

https://github.com/pytorch/vision/blob/c00c4d36d07552dbe0cc4fd399a9bf911bf99fcb/test/datasets_utils.py#L435-L437

But there are also tests that do not specify a config, which means the `DEFAULT_CONFIG` is used as is:

https://github.com/pytorch/vision/blob/c00c4d36d07552dbe0cc4fd399a9bf911bf99fcb/test/datasets_utils.py#L431-L433

---

The `create_dataset()` method does not simply do

```python
if config is None:
     config = cls.DEFAULT_CONFIG.copy()
```

because we want to make sure, that we always get a "complete" config in `dataset_args` and `inject_fake_data`. 

Imagine a dataset with multiple keyword arguments. We could obtain a complete config by checking all available combinations. Since the number of tests would grow quadratic with the number of possible values for the arguments this should not be they way to go. Thus, we populate the `DEFAULT_CONFIG` with complete set of parameters and only update the ones the user wants to test.

Reply:
IIUC the last changes:

- `DEFAULT_CONFIG` serves both as a "default" config when `CONFIGS` is None, and also as a fall-back for default values for parameters that are not specified a `CONFIGS` entry. This second use-case doesn't seem to be explicitly documented
- if `DEFAULT_CONFIG` is left to None, it will be set to the default parameter values for all keyword arguements, from the constructor. This means that when a dataset has a non-keyword argument, `DEFAULT_CONFIG` *must* be set.

Is this correct?